engine/dumps/bank02.asm

```diff
@@ -0,0 +1,3164 @@
+	ld a, [hl]
+	cp $ff
+	jr nz, .sub_832c
+	ld hl, $0002
+	add hl, bc
+	ld a, [hl]
+	add $01
+	sub d
+	jr c, .sub_832c
+	cp $0b
+	jr nc, .sub_832c
+	push de
+	push bc
+	call Function80eb
+	pop bc
+	pop de
+.sub_832c
+	ld hl, $0010
+	add hl, bc
+	ld b, h
+	ld c, l
+	ldh a, [hConnectionStripLength]
+	inc a
+	cp $10
+	jr nz, .sub_82fc
+	ret
+
+Function833a: ; 02:433a
+	ld a, c
+	push af
+	call InitMovementBuffer
+	ld a, $29
+	call AppendToMovementBuffer
+	ld a, b
+	call GetMapObject
+	ld hl, $0000
+	add hl, bc
+	ld a, [hl]
+	cp $ff
+	jr z, .sub_8361
+	call GetObjectStruct
+	ld hl, $0011
+	add hl, bc
+	ld a, [hl]
+	ld hl, $0010
+	add hl, bc
+	ld b, [hl]
+	ld c, a
+	jr .sub_836c
+.sub_8361
+	ld hl, $0002
+	add hl, bc
+	ld a, [hl]
+	ld hl, $0003
+	add hl, bc
+	ld b, [hl]
+	ld c, a
+.sub_836c
+	pop af
+	call ComputePathToWalkToPlayer
+	ld a, $28
+	call AppendToMovementBuffer
+	ld a, $32
+	call AppendToMovementBuffer
+	xor a
+	ret
+
+Function837c: ; 02:437c
+	call InitMovementBuffer
+	push bc
+	ld a, b
+	call GetMapObject
+	ld hl, $0000
+	add hl, bc
+	ld a, [hl]
+	call GetObjectStruct
+	ld hl, $0010
+	add hl, bc
+	ld a, [hl]
+	ld hl, $0011
+	add hl, bc
+	ld c, [hl]
+	ld b, a
+	pop hl
+	ld a, l
+	call ComputePathToWalkToPlayer
+	ld a, $32
+	call AppendToMovementBuffer
+	ret
+
+Function83a2: ; 02:43a2
+	push de
+	call InitMovementBuffer
+	pop de
+	call Function83b0
+	ld a, $32
+	call AppendToMovementBuffer
+	ret
+
+Function83b0: ; 02:43b0
+	push de
+	push bc
+	ld a, c
+	call GetMapObject
+	ld hl, $0000
+	add hl, bc
+	ld a, [hl]
+	call GetObjectStruct
+	ld d, b
+	ld e, c
+	pop bc
+	ld a, b
+	call GetMapObject
+	ld hl, $0000
+	add hl, bc
+	ld a, [hl]
+	call GetObjectStruct
+	ld hl, $0010
+	add hl, bc
+	ld a, [hl]
+	ld hl, $0011
+	add hl, bc
+	ld c, [hl]
+	ld b, a
+	ld hl, $0010
+	add hl, de
+	ld a, [hl]
+	ld hl, $0011
+	add hl, de
+	ld e, [hl]
+	ld d, a
+	pop af
+	call ComputePathToWalkToPlayer
+	ret
+
+Function83e8: ; 02:43e8
+	ld hl, wcb70
+	push hl
+	ld a, [hl]
+	ld l, a
+	ld h, $00
+	ld de, Table83fb
+	add hl, hl
+	add hl, de
+	ld e, [hl]
+	inc hl
+	ld d, [hl]
+	pop hl
+	push de
+	ret
+
+Table83fb: ; 02:43fb
+	dw Function8432
+	dw Function844a
+	dw Function8459
+	dw Function8433
+	dw Function8438
+	dw Function843d
+	dw Function8468
+	dw Function8477
+	dw Function8485
+	dw Function848b
+	dw Function8494
+	dw Function849b
+	dw Function84a9
+	dw Function84af
+	dw Function84b8
+
+Function8419: ; 02:4419
+	ld a, c
+	ld [wVBCopyFarSrcBank], a
+	ld a, l
+	ld [wVBCopyFarSrc], a
+	ld a, h
+	ld [wVBCopyFarSrc+1], a
+
+Function8425: ; 02:4425
+	ld a, e
+	ld [wVBCopyFarDst], a
+	ld a, d
+	ld [wVBCopyFarDst+1], a
+	
+Function842d: ; 02:442d
+	ld a, b
+	ld [wVBCopyFarSize], a
+	ret
+
+Function8432: ; 02:4432
+	ret
+
+Function8433: ; 02:4433
+	ld hl, ShockEmoteGFX
+	jr Function8440
+
+Function8438: ; 02:4438
+	ld hl, QuestionEmoteGFX
+	jr Function8440
+
+Function843d: ; 02:443d
+	ld hl, HappyEmoteGFX
+
+Function8440: ; 02:4440
+	ld de, vChars1 + $780
+	ld b, $04
```

Comment:
Many missed references to hardcoded banks. (`ld b, $nn` for video copy functions)

Reply:
b is the size of the gfx, c is the bank